ocaml/xapi/nm.ml

```diff
@@ -45,45 +45,41 @@ let determine_mtu pif_rc net_rc =
 
 let determine_ethtool_settings properties oc =
   let proc key =
-    if List.mem_assoc ("ethtool-" ^ key) oc then
-      let value = List.assoc ("ethtool-" ^ key) oc in
-      if value = "true" || value = "on" then
+    match
+      (List.assoc_opt ("ethtool-" ^ key) oc, List.assoc_opt key properties)
+    with
+    | Some value, _ when value = "true" || value = "on" ->
```

Comment:
`Some ("on"|"true")` would be shorter and slightly more efficient

Reply:
Oh, I didn't know that was possible :)